### PR TITLE
Raising RuntimeError when trying to get empty body encoding

### LINKED ISSUE #4214: ClientResponse.get_encoding() throws TypeError if it hasn't been read
## Long story short
`ClientResponse.get_encoding()` throws `TypeError: object of type 'NoneType' has no len()` if we don't first call `text()` or `json()` to read from it, i.e. `_body is None`

Related: #4112 

## Expected behaviour
Does not throw an error, returns some default value

## Actual behaviour
```python-traceback
Traceback (most recent call last):
  File "/home/transfusion/aiohttp/aiohttp_issue_4112.py", line 14, in <module>
    loop.run_until_complete(test())
  File "/usr/lib/python3.7/asyncio/base_events.py", line 579, in run_until_complete
    return future.result()
  File "/home/transfusion/aiohttp/aiohttp_issue_4112.py", line 10, in test
    print(r.get_encoding())
  File "/home/transfusion/aiohttp/aiohttp/client_reqrep.py", line 942, in get_encoding
    encoding = chardet.detect(self._body)['encoding']
  File "/home/transfusion/python37/lib/python3.7/site-packages/cchardet/__init__.py", line 15, in detect
    encoding, confidence = _cchardet.detect_with_confidence(msg)
  File "src/cchardet/_cchardet.pyx", line 15, in cchardet._cchardet.detect_with_confidence
TypeError: object of type 'NoneType' has no len()
```
## Steps to reproduce

Find any webpage that doesn't include a `charset` parameter in its `Content-Type` response header; you may check headers only with `curl -v -s http://example.com 1> /dev/null`

```py
import aiohttp
import asyncio


async def test():

    async with aiohttp.ClientSession() as session:
        r = await session.get('http://www.africau.edu', ssl=None)
        # print(await r.text())
        print(r.get_encoding())

if __name__ == "__main__":
    loop = asyncio.get_event_loop()
    loop.run_until_complete(test())
```

## Your environment

4.0.0a1, Commit f356eeb8ccefd1e176536d396d026a23da1f1695

### PR DESCRIPTION
**Pull request by Purusah**
## What do these changes do?

Function get_encoding raise RuntimeError when user is trying to read not present body and header doesn't provide info about encoding

## Are there changes in behavior for the user?

Make error more verbose

## Related issue number

Fixes #4214

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: CONTRIBUTORS.txt [hunk 1]
```diff
@@ -268,6 +268,7 @@ Vladimir Rutsky
 Vladimir Shulyak
 Vladimir Zakharov
 Vladyslav Bondar
+Vladyslav Bohaichuk
```
Comment (webknjaz) [inline CONTRIBUTORS.txt:270]: this should be one line up

File: docs/client_reference.rst [hunk 2]
```diff
@@ -1305,6 +1305,8 @@ Response object
       decode a response. Some encodings detected by cchardet are not known by
       Python (e.g. VISCII).
 
+      :raise RuntimeError: if body hasn't been read for :term:`cchardet` usage
```
Comment (webknjaz) [inline docs/client_reference.rst]:
```suggestion
      :raise RuntimeError: if called before the body has been read, for :term:`cchardet` usage
```

File: aiohttp/client_reqrep.py [hunk 3]
```diff
@@ -938,6 +938,8 @@ def get_encoding(self) -> str:
             if mimetype.type == 'application' and mimetype.subtype == 'json':
                 # RFC 7159 states that the default encoding is UTF-8.
                 encoding = 'utf-8'
+            elif self._body is None:
+                raise RuntimeError("Trying to get encoding of not read body")
```
Comment (webknjaz) [inline aiohttp/client_reqrep.py]:
```suggestion
                raise RuntimeError('Cannot guess the encoding of a not yet read body')
```

File: tests/test_client_response.py [hunk 4]
```diff
@@ -424,6 +424,30 @@ def side_effect(*args, **kwargs):
...
+    def side_effect(*args, **kwargs):
+        fut = loop.create_future()
+        fut.set_result('{"encoding": "test"}')
+        return fut
+
+    response._headers = {'Content-Type': 'text/html'}
+    content = response.content = mock.Mock()
+    content.read.side_effect = side_effect
+
+    with pytest.raises(RuntimeError):
```
Comment (webknjaz) [inline tests/test_client_response.py]:
```suggestion
    with pytest.raises(
        RuntimeError,
        match='^Cannot guess the encoding of a not yet read body$',
    ):
```

Comment (Purusah) [reply tests/test_client_response.py]: Updated PR, thank's for review
Does it always make sense to check exception message?

**Comment by asvetlov** [comment]
Thanks!

**Pull request merged**